src/Dependencies/Collections/ImmutableSegmentedDictionary`2.cs

```diff
@@ -0,0 +1,360 @@
+            var dictionary = new SegmentedDictionary<TKey, TValue>(self._dictionary, self._dictionary.Comparer);
+            foreach (var pair in pairs)
+            {
+                dictionary.Add(pair.Key, pair.Value);
+            }
+
+            return new ImmutableSegmentedDictionary<TKey, TValue>(dictionary);
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> Clear()
+        {
+            var self = this;
+            if (self.IsEmpty)
+            {
+                return self;
+            }
+
+            return Empty.WithComparer(self.KeyComparer);
+        }
+
+        public bool Contains(KeyValuePair<TKey, TValue> pair)
+        {
+            return TryGetValue(pair.Key, out var value)
+                && EqualityComparer<TValue>.Default.Equals(value, pair.Value);
+        }
+
+        public bool ContainsKey(TKey key)
+            => _dictionary.ContainsKey(key);
+
+        public bool ContainsValue(TValue value)
+            => _dictionary.ContainsValue(value);
+
+        public Enumerator GetEnumerator()
+            => new(_dictionary, Enumerator.ReturnType.KeyValuePair);
+
+        public ImmutableSegmentedDictionary<TKey, TValue> Remove(TKey key)
+        {
+            var self = this;
+            if (!self._dictionary.ContainsKey(key))
+                return self;
+
+            var dictionary = new SegmentedDictionary<TKey, TValue>(self._dictionary, self._dictionary.Comparer);
+            dictionary.Remove(key);
+            return new ImmutableSegmentedDictionary<TKey, TValue>(dictionary);
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> RemoveRange(IEnumerable<TKey> keys)
+        {
+            if (keys is null)
+                throw new ArgumentNullException(nameof(keys));
+
+            var result = ToBuilder();
+            result.RemoveRange(keys);
+            return result.ToImmutable();
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> SetItem(TKey key, TValue value)
+        {
+            var self = this;
+            var dictionary = new SegmentedDictionary<TKey, TValue>(self._dictionary, self._dictionary.Comparer);
+            dictionary[key] = value;
+            return new ImmutableSegmentedDictionary<TKey, TValue>(dictionary);
+        }
+
+        public ImmutableSegmentedDictionary<TKey, TValue> SetItems(IEnumerable<KeyValuePair<TKey, TValue>> items)
+        {
+            if (items is null)
+                throw new ArgumentNullException(nameof(items));
+
+            var result = ToBuilder();
+            foreach (var item in items)
+            {
+                result[item.Key] = item.Value;
+            }
+
+            return result.ToImmutable();
+        }
+
+#pragma warning disable CS8767 // Nullability of reference types in type of parameter doesn't match implicitly implemented member (possibly because of nullability attributes).
+        public bool TryGetKey(TKey equalKey, [NotNullWhen(true)] out TKey? actualKey)
```

Comment:
>NotNullWhen [](start = 46, length = 11)

I'd rather make this attribute conditional on NETCOREAPP.

Reply:
Should also remove this from the csproj file. 
```
<ItemGroup>
    <!-- ⚠ Consuming projects outside dotnet/roslyn will be required to manually reproduce the APIs for any linked files added here -->
    <Compile Include="..\..\Compilers\Core\Portable\InternalUtilities\NullableAttributes.cs" Link="NullableAttributes.cs" />
  </ItemGroup>
```

---
In reply to: [561158696](https://github.com/dotnet/roslyn/pull/50189#discussion_r561158696) [](ancestors = 561158696)

Reply:
➡️ See https://github.com/dotnet/roslyn/pull/50189#discussion_r561163187

Reply:
It looks like this was a mistake in the implementation. When this method returns `false`, `actualKey` is expected to be set to `equalKey`.

Reply:
➡️ Implementation fixed and suppression removed